### ISSUE TITLE
Feature Request: synchronous flag for `hab svc <start/stop>`

### ISSUE THREAD

**Issue by smartb-pair** [habitat-sh/habitat#2680]
Inspiration for this feature comes from a flag used by `svcadm` on Illumos and Solaris distros, `-s`, which tells the service state change to be synchronous:
```
If the -s option is specified, svcadm disables each service instance and then waits for each service instance to enter the disabled state. svcadm will return early if it determines that the service cannot reach this state without administrator intervention.
```
Such a flag for `hab svc <start/stop>` would be extremely useful for things like Test Kitchen integration tests, orchestration with tools like Chef, Ansible, etc, and operator intervention.

**Comment by chrisortman** [comment]
I think this could also be useful when doing package development inside the studio. I <3 the the supervisor is already running and I can startup a background (database) process to bind to, but would be nice to do `hab start` see what I did wrong and `ctrl+c` to be done vs `hab start && sup-log` .. `hab svc unload`

**Comment by smartb-pair** [comment]
We keep finding more and more reasons we need synchronous start/stop/restart in orchestration and scripting tools - it's even proving difficult to judge when a service is up or down using `hab svc status <svcname>`.

**Comment by eeyun** [comment]
Yeah this is an excellent idea :) we're most definitely going to want a first class cli experience for this kind of thing.

**Comment by smartb-pair** [comment]
Upon further thought, we think it actually makes more sense for 'synchronous' mode to be the _default_, and async to be a flag.

**Comment by smartb-pair** [comment]
...aaaand just ran into this again today. Just confirms again that having `hab svc start` return immediately (when a package may be installing/starting in the background) makes programming with `hab` quite difficult/annoying :)

**Comment by bixu** [comment]
This would still be a big win for CI flows.

**Comment by themightychris** [comment]
I'm constantly copy-pasting code from my health check hooks into sleep loops in CI flows, we need an easy command that leverages supervisor plumbing for start-service-and-wait-for-health
